tools/updatectl/hub.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"path/filepath"
+)
+
+const (
+	baseHubURL = "https://hub.grid.tf/api/flist"
+)
+
+// Hub API
+type Hub struct {
+	base   *url.URL
+	client http.Client
+}
+
+func newTestHub(baseURL, user, token string) (*Hub, error) {
```

Comment:
can you most this into the test file so we don't pollute actual code with testing code.

Reply:
yup